contracts/core/OTCWrapper.sol

```diff
@@ -732,6 +756,54 @@ contract OTCWrapper is Initializable, OwnableUpgradeable, ReentrancyGuardUpgrade
         emit VaultSettled(_orderID);
     }
 
+    /**
+     * @notice allows user to redeem after expiry
+     * @dev can only be called by the user who is the order buyer
+     * @param _orderID order id
+     */
+    function redeem(uint256 _orderID) external nonReentrant {
+        require(orderStatus[_orderID] == OrderStatus.Succeeded, "OTCWrapper: inexistent or unsuccessful order");
+
+        Order memory order = orders[_orderID];
+
+        require(order.buyer == _msgSender(), "OTCWrapper: sender is not the order buyer");
+
+        // check if vault has enough collateral to cover payout
+        (UtilsWrapperInterface.Vault memory vault, uint256 typeVault, ) = controller.getVaultWithDetails(
+            address(this),
+            order.vaultID
+        );
+
+        (, bool isValidVault) = calculator.getExcessCollateral(vault, typeVault);
+
+        // require that vault is valid (has excess collateral) before redeeming
+        // to avoid allowing redeeming undercollateralized vaults
+        require(isValidVault, "OTCWrapper: insuficient collateral to redeem");
+
+        // transfer otokens to redeem - an approve() by the _msgSender() is required beforehand
+        IERC20(order.oToken).safeTransferFrom(_msgSender(), address(this), order.size);
+
+        UtilsWrapperInterface.ActionArgs[] memory actions = new UtilsWrapperInterface.ActionArgs[](1);
+
+        actions[0] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.Redeem,
+            address(0), // not used
+            order.buyer, // address to transfer to
+            order.oToken, // otoken address
+            0, // not used
+            order.size, // not used
```

Comment:
order.size is used - lets remove the `// not used` comment

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/5/commits/b027f461b5643aa142337ff48046a09b22757b7f